// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: list_tasks_by_source.proto

package evaluatord

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"strings"
	"time"
	"unicode/utf8"

	"github.com/golang/protobuf/ptypes"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = ptypes.DynamicAny{}
)

// define the regex for a UUID once up-front
var _list_tasks_by_source_uuidPattern = regexp.MustCompile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$")

// Validate checks the field values on ListTasksBySourceRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *ListTasksBySourceRequest) Validate() error {
	if m == nil {
		return nil
	}

	if m.GetSource() == nil {
		return ListTasksBySourceRequestValidationError{
			field:  "Source",
			reason: "value is required",
		}
	}

	if v, ok := interface{}(m.GetSource()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return ListTasksBySourceRequestValidationError{
				field:  "Source",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if v, ok := interface{}(m.GetRangeFrom()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return ListTasksBySourceRequestValidationError{
				field:  "RangeFrom",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	if v, ok := interface{}(m.GetRangeTo()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return ListTasksBySourceRequestValidationError{
				field:  "RangeTo",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	return nil
}

// ListTasksBySourceRequestValidationError is the validation error returned by
// ListTasksBySourceRequest.Validate if the designated constraints aren't met.
type ListTasksBySourceRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListTasksBySourceRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListTasksBySourceRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListTasksBySourceRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListTasksBySourceRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListTasksBySourceRequestValidationError) ErrorName() string {
	return "ListTasksBySourceRequestValidationError"
}

// Error satisfies the builtin error interface
func (e ListTasksBySourceRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListTasksBySourceRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListTasksBySourceRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListTasksBySourceRequestValidationError{}

// Validate checks the field values on ListTasksBySourceResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *ListTasksBySourceResponse) Validate() error {
	if m == nil {
		return nil
	}

	for idx, item := range m.GetTasks() {
		_, _ = idx, item

		if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return ListTasksBySourceResponseValidationError{
					field:  fmt.Sprintf("Tasks[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	return nil
}

// ListTasksBySourceResponseValidationError is the validation error returned by
// ListTasksBySourceResponse.Validate if the designated constraints aren't met.
type ListTasksBySourceResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListTasksBySourceResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListTasksBySourceResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListTasksBySourceResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListTasksBySourceResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListTasksBySourceResponseValidationError) ErrorName() string {
	return "ListTasksBySourceResponseValidationError"
}

// Error satisfies the builtin error interface
func (e ListTasksBySourceResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListTasksBySourceResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListTasksBySourceResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListTasksBySourceResponseValidationError{}
